rs/messaging/src/routing/stream_handler.rs

```diff
@@ -616,14 +624,15 @@ impl StreamHandlerImpl {
                         state,
                         available_guaranteed_response_memory,
                     ) {
-                        None => {
+                        Ok(maybe_lost_cycles) => {
                             // Reject response successfully inducted or dropped.
+                            lost_cycles += maybe_lost_cycles;
```

Comment:
Are there any things to be considered in terms of overflowswhen using the cycles type?

Reply:
My main concern was related to surprising panics when an overflow/underflow happens. We have had cases where wrapper types introduced crashes on underflows (I think it was something related to `Duration`). I was wondering whether something similar could happen with cycles?

Reply:
Looking at the implementations of `Add` / `AddAsign` / `Sub` / `SubAssign` / `Mul` for `Cycles`, they all use `saturating_<op>` internally. Switching to anything else without going through the code with a fine toothed comb seems like a recipe to shoot oneself in the foot. IOW, it's unrealistic to expect any changes to `Cycles` arithmetic operations.